packages/story-editor/src/components/panels/design/textStyle/stylePresets/stylePresets.js

```diff
@@ -0,0 +1,188 @@
+    height: 32px;
+    width: 32px;
+    margin-left: -4px;
+  }
+`;
+
+const NoStylesWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: calc(100% + 32px);
+  margin-left: -16px;
+  background-color: ${({ theme }) => theme.colors.opacity.black24};
+  height: 64px;
+`;
+
+const NoStylesText = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+`;
+
+function PresetPanel({ pushUpdate }) {
+  const { globalStoryStyles } = useStory(
+    ({
+      state: {
+        story: { globalStoryStyles },
+      },
+    }) => {
+      return {
+        globalStoryStyles,
+      };
+    }
+  );
+
+  const {
+    refs: { inspector },
+  } = useInspector();
+  const buttonRef = useRef(null);
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+
+  const { textStyles } = globalStoryStyles;
+  const hasPresets = textStyles.length > 0;
+
+  const handleApplyStyle = useApplyStyle({ pushUpdate });
+  const { addGlobalPreset } = useAddPreset({ presetType: PRESET_TYPES.STYLE });
+
+  const handlePresetClick = (preset) => {
+    handleApplyStyle(preset);
+  };
+
+  const spacing = useMemo(() => ({ x: 20 }), []);
+  return (
+    <>
+      <PresetsHeader>
+        <SubHeading size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {__('Recent Saved Styles', 'web-stories')}
+        </SubHeading>
+        <StyledButton
+          type={BUTTON_TYPES.TERTIARY}
+          size={BUTTON_SIZES.SMALL}
+          variant={BUTTON_VARIANTS.SQUARE}
+          onClick={addGlobalPreset}
+          aria-label={__('Add style', 'web-stories')}
+        >
+          <Icons.Plus />
+        </StyledButton>
+      </PresetsHeader>
+      {hasPresets && (
+        <>
+          <StylesWrapper>
+            <StyleGroup
+              styles={textStyles.slice(-2)}
+              handleClick={handlePresetClick}
+            />
+          </StylesWrapper>
+          <StyledMoreButton
+            ref={buttonRef}
+            type={BUTTON_TYPES.PLAIN}
+            size={BUTTON_SIZES.SMALL}
+            variant={BUTTON_VARIANTS.RECTANGLE}
+            onClick={() => setIsPopupOpen(true)}
+          >
+            {__('More styles', 'web-stories')}
+            <Icons.ChevronDownSmall />
+          </StyledMoreButton>
+          <Popup
+            anchor={buttonRef}
+            dock={inspector}
+            isOpen={isPopupOpen}
+            placement={PLACEMENT.LEFT_START}
+            spacing={spacing}
+            renderContents={() => (
+              <StyleManager
+                styles={textStyles}
+                applyStyle={handlePresetClick}
+                onClose={() => setIsPopupOpen(false)}
+              />
+            )}
+          />
+        </>
+      )}
+      {!hasPresets && (
+        <NoStylesWrapper>
+          <NoStylesText size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+            {__('No Styles Saved', 'web-stories')}
+          </NoStylesText>
+        </NoStylesWrapper>
```

Comment:
Is this in Figma? It looks a bit wrong - like there's a hole in the panel because of the difference in BG color:

![Screen Shot 2021-11-08 at 16 12 32](https://user-images.githubusercontent.com/637548/140818934-0ae1dc9f-3955-4cea-ad16-935bbcdbc48e.png)

Reply:
[Figma](https://www.figma.com/file/dc9xrVlRhQq4KTE9WQzU1R/_Editor---Sprint-Feature-Level-Designs_?node-id=10250%3A436112). The Alignment panel also has this background.

cc @agingoldseco for thoughts.